packages/flutter_android_lifecycle/android/src/main/java/io/flutter/embedding/engine/plugins/lifecycle/FlutterLifecycleAdapter.java

```diff
@@ -0,0 +1,46 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.embedding.engine.plugins.lifecycle;
+
+import androidx.annotation.NonNull;
+import androidx.lifecycle.Lifecycle;
+import io.flutter.Log;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+public class FlutterLifecycleAdapter {
+  private static final String TAG = "FlutterLifecycleAdapter";
+
+  @NonNull private Lifecycle lifecycle = null;
+
+  public FlutterLifecycleAdapter(@NonNull Object reference) {
+    try {
+      Class hiddenLifecycleClass =
+          Class.forName("io.flutter.embedding.engine.plugins.lifecycle.HiddenLifecycleReference");
+
+      if (!reference.getClass().equals(hiddenLifecycleClass)) {
+        throw new IllegalArgumentException(
+            "The reference argument must be of type HiddenLifecycleReference. Was actually "
+                + reference);
+      }
+
+      Method getLifecycle = reference.getClass().getMethod("getLifecycle");
+      this.lifecycle = (Lifecycle) getLifecycle.invoke(reference);
+    } catch (ClassNotFoundException
+        | NoSuchMethodException
+        | IllegalAccessException
+        | InvocationTargetException e) {
+      Log.w(
```

Comment:
If we hit this case, `getLifecycle()` will return null. Should we maybe allow all those exceptions to actually throw instead of catching them?

Reply:
I think null returning null is reasonable, especially as we expect that to be the case on stable for awhile.

I added some comments on getLifecycle and changed it to Nullable.